default/static/js/uploader.js

```diff
@@ -81,6 +81,10 @@ var app = {};
         addFile(data);
         if (!isUploadButtonEnabled()) {
             enableUploadButton();
+            // if UploadButton was disabled in split-packages, check split size
+            if ($(location).attr('pathname') === '/split-packages/') {
+                $.event.trigger('appHookCheckSplitSize');
```

Comment:
The best place to check this would be the Start button click listener, the value of the input can change after a file is added to the fileupload box.

Reply:
The Start button is disable if the input is invalid so we need two checks, one in `split-packages.html` that checks every time something is typed into the input box, and another here because when a file is added to the fileupload box the disable Start button always get enabled with `enableUploadButton()`.

Reply:
I guess what you are saying here is that the Start button should be disabled if any input is invalid. I agree with this, however, the solution is spread around and it doesn't try to work along with the code in `uploader.js`. The intention of this file is to gather all the common code between the operations in Toucan so when we add a new one, most of the code can be reused.

Note that the placement of this code is not the best one: the Start button is enabled, and then an event that may disable the button again is triggered. The `check_split_size` function enables the button without considering if there is a file selected or not, which makes it possible to enable the button without having any file (try this: choose a file, then delete it and edit the splitSize input).

Please define a function in `upload.js` that runs a list of validators and enables/disables the button accordingly. Note that `upload.js` keeps an internal list of files to be sent; the function must validate that this list is not empty as well. You can keep a list of validators (functions) as a variable, and have another function to add validators to the list that can be accessed from the outside. The validation function should be visible as well, so you would be able to call it for certain events.

Reply:
Done. There was no need to validate that the list is not empty in the function created for the input of split-size because the start button is not disabled there.